src/emcee/ensemble.py

```diff
@@ -147,8 +158,7 @@ def __init__(
 
         # This is a random number generator that we can easily set the state
         # of without affecting the numpy-wide generator
-        self._random = np.random.mtrand.RandomState()
-        self._random.set_state(state)
+        self._random = _check_random_state(seed, state)
```

Comment:
Does the initialisation of the `EnsembleSampler` work  if an initialized `backend` is provided, and the backend was created using a `np.random.Generator`?

Reply:
Not sure, I'll check and get back to you, from a quick look it will probably not work